tests/test_collection_metadata.py

```diff
@@ -0,0 +1,513 @@
+        """Test CMR reader handles uppercase environment names."""
+        reader_uat = CollectionMetadataReader(environment="UAT")
+        assert reader_uat.environment == "uat"
+        assert reader_uat.provider == "NSIDC_CUAT"
+
+        reader_prod = CollectionMetadataReader(environment="PROD")
+        assert reader_prod.environment == "prod"
+        assert reader_prod.provider == "NSIDC_CPRD"
+
+    @patch("earthaccess.login")
+    @patch("earthaccess.search_datasets")
+    def test_get_collection_metadata_success(
+        self,
+        mock_search,
+        mock_login,
+        collection_metadata_reader_uat,
+        sample_ummc_response,
+    ):
+        """Test successful collection metadata retrieval."""
+        # Setup mocks
+        mock_login.return_value = True
+        mock_search.return_value = sample_ummc_response
+
+        # Call the method
+        metadata = collection_metadata_reader_uat.get_collection_metadata(
+            "SNEX23_SSADUCk", "1"
+        )
+
+        # Verify the results
+        assert isinstance(metadata, CollectionMetadata)
+        assert metadata.short_name == "SNEX23_SSADUCk"
+        assert metadata.version == "1"
+        assert (
+            metadata.entry_title
+            == "SnowEx23 Snow Sensor Array Duck Pass Time Series V001"
+        )
+        assert metadata.granule_spatial_representation == "GEODETIC"
+        assert metadata.processing_level_id == "3"
+        assert metadata.collection_data_type == "SCIENCE_QUALITY"
+
+        # Check spatial extent
+        assert metadata.spatial_extent is not None
+        assert len(metadata.spatial_extent) == 1
+        assert metadata.spatial_extent[0]["WestBoundingCoordinate"] == -119.5
+
+        # Check temporal extent
+        assert metadata.temporal_extent is not None
+        assert len(metadata.temporal_extent) == 1
+        assert (
+            metadata.temporal_extent[0]["BeginningDateTime"]
+            == "2023-01-01T00:00:00.000Z"
+        )
+
+        # Check raw UMM-C is preserved
+        assert metadata.raw_ummc is not None
+        assert metadata.raw_ummc["ShortName"] == "SNEX23_SSADUCk"
+
+        # Verify mocks were called correctly
+        mock_login.assert_called_once_with(
+            strategy="environment", system=earthaccess.UAT
+        )
+        mock_search.assert_called_once_with(
+            short_name="SNEX23_SSADUCk",
+            version="1",
+            has_granules=None,
+            provider="NSIDC_CUAT",
+        )
+
+    @patch("earthaccess.login")
+    def test_get_collection_metadata_login_failure(
+        self, mock_login, collection_metadata_reader_uat
+    ):
+        """Test handling of Earthdata login failure."""
+        mock_login.return_value = False
+
+        with pytest.raises(Exception) as exc_info:
+            collection_metadata_reader_uat.get_collection_metadata("TEST", "1")
+
+        assert "Earthdata login failed" in str(exc_info.value)
```

Comment:
I need to know why!

Reply:
See other comment--the `earthaccess` exception has the deets.